shared/Microsoft.Extensions.ActivatorUtilities.Sources/ActivatorUtilities.cs

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+
+namespace Microsoft.Extensions.Internal
```

Comment:
Could we use `ifdefs` to replace https://github.com/aspnet/DependencyInjection/blob/dev/src/DI.Abstractions/DIActivatorUtilities.cs? Something like https://github.com/aspnet/Common/blob/dev/shared/Microsoft.Extensions.WebEncoders.Sources/WebEncoders.cs#L18?